core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -264,12 +264,18 @@ KEEP_INIT _start
 		cmp	r0, #CFG_TEE_CORE_NB_CORE
 		/* Unsupported CPU, park it before it breaks something */
 		bge	unhandled_cpu
-		ldr	r1, =stack_tmp_stride
-		ldr	r1, [r1]
+
+		adr	r3, stack_tmp_stride_rel
+		ldr	r1, [r3]
+		ldr	r1, [r1, r3]
```

Comment:
I don't catch this trick. Shouldn't it be `add r1, r1, r3` at lien 270? (and equiv at line 275)

Reply:
OK, I'll add a comment. At the address of `stack_tmp_stride_rel`, is the offset from that address to `__stack_tmp_stride` stored. `add r1, r1, r3` would compute the address of `__stack_tmp_stride`, while `ldr r1, [r1, r3]` does a load from that address.

Reply:
oh, right!